### PR TITLE
[NO SQUASH] Auth-related fixes

### LINKED ISSUE #11877: New player causes: duplicate key value violates unique constraint "auth_name_key"
##### Minetest version
```
mt 5.4.1
```

##### OS / Hardware

Operating system: debian 11
CPU: AMD Epyc 7282
Postgres postgresql-13  13.5-0+deb11u1

##### Summary
```
2021-12-19 11:56:00: ERROR[Main]: ServerError: AsyncErr: ServerThread::run Lua: Runtime error from mod '' in callback createAuth(): PostgreSQL database error: ERROR: duplicate key value violates unique constraint "auth_name_key"
2021-12-19 11:56:00: ERROR[Main]: DETAIL: Key (name)=(basu) already exists.
2021-12-19 11:56:00: ERROR[Main]: stack traceback:
2021-12-19 11:56:00: ERROR[Main]: [C]: at 0x557f09c49420
```
Backup of 2021-12-16-19-29 does not have a player called "basu" in the auth table, now there is one but logindate is -1

There is a different playername that includes "basu" , like Xyzbasu

##### Repro

I do not have a repro of any kind, the error happened over the course of 2 years more than once, while the other 16 000 accs had no trouble of that kind.

For my reference 1189

### LINKED ISSUE #12233: disallow_empty_password = true ... still allow client to change later to empty, what I don't want !
##### Summary
the minetest.cond setting 'disallow_empty_password = true' allow server admin to block client without a password, or deny empty password,
BUT after join game, gamer can set in client a new password that is against my set rule no empty.

##### Steps to reproduce
login, press esc, press change password ...

IF we take the willing or rule from admin serious
set disallow_empty_password = true
then also a change from client / gamer must be checked over this rule and revoke if true and new password empty !

### PR DESCRIPTION
**Pull request by sfan5**
fixes #11877 and #12220 (the crash everyone has been complaining about)
fixes #12233 (an ordinary bug)
and cleans up some code

## To do

This PR is Ready for Review.

## How to test

Exercise left to the reader.

### CODE REVIEW

File: src/network/networkprotocol.h [hunk 1]
```diff
@@ -1027,20 +1027,20 @@ enum NetProtoCompressionMode {
 	NETPROTO_COMPRESSION_NONE = 0,
 };
 
-const static std::string accessDeniedStrings[SERVER_ACCESSDENIED_MAX] = {
+static const char *accessDeniedStrings[SERVER_ACCESSDENIED_MAX] = {
```
Comment (fluxionary) [inline src/network/networkprotocol.h:1030]: When I build, I get spammed with
```
/opt/minetest-vanilla/src/network/networkprotocol.h:1030:20: warning: ‘accessDeniedStrings’ defined but not used [-Wunused-variable]
 1030 | static const char *accessDeniedStrings[SERVER_ACCESSDENIED_MAX] = {
``` 

though I see it's used here:
https://github.com/minetest/minetest/blob/2500eabcf20ea650f4a7bf0903a9d5dfc2bcf87f/src/network/clientpackethandler.cpp#L202-L225

**Comment by fluxionary** [comment]
Haven't been able to crash it so far, good job :+1:

File: src/network/networkprotocol.h (hunk 1 above)
Comment (sfan5) [reply src/network/networkprotocol.h]: Not sure why it didn't happen locally but I changed it back :shrug:

**Review by SmallJoker — APPROVED** [review]
`disallow_empty_password = true`
:ballot_box_with_check: Join with empty password (existing player)
:ballot_box_with_check: Failed attempt to register with empty password
:ballot_box_with_check: Failed password change to empty (after rejoin)
:ballot_box_with_check: Successful password change to non-empty

`disallow_empty_password = false`
:ballot_box_with_check: Successful empty password registration
:ballot_box_with_check: Successful password change to empty
:ballot_box_with_check: Successful password change to non-empty

Code looks good too. :+1:

**Pull request merged**